emulator: Unit testing for examples got panic when the header with blank value provided

<!-- PLEASE FILL IN THE FOLLOWINGS for EVERY ISSUE you make. -->

## Describe the bug / error

Those requests that have headers that have blank values will cause panic.

reproductive branch: https://github.com/musaprg/proxy-wasm-go-sdk/tree/reproduce-http-blank-header-panic

I modified `http_headers` example and call `proxywasm.GetHttpRequestHeader("empty-key")` to obtain the header value which is blank. Then, I got the panic message below.

```
$ make test
go test -tags=proxytest github.com/tetratelabs/proxy-wasm-go-sdk/examples/dispatch_call_on_tick github.com/tetratelabs/proxy-wasm-go-sdk/examples/foreign_call_on_tick github.com/tetratelabs/proxy-wasm-go-sdk/exam
ples/helloworld github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_auth_random github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_body github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_headers github
.com/tetratelabs/proxy-wasm-go-sdk/examples/http_routing github.com/tetratelabs/proxy-wasm-go-sdk/examples/metrics github.com/tetratelabs/proxy-wasm-go-sdk/examples/network github.com/tetratelabs/proxy-wasm-go-sd
k/examples/shared_data github.com/tetratelabs/proxy-wasm-go-sdk/examples/shared_queue/receiver github.com/tetratelabs/proxy-wasm-go-sdk/examples/shared_queue/sender github.com/tetratelabs/proxy-wasm-go-sdk/exampl
es/vm_plugin_configuration github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/internal github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/proxytest github.com/te
tratelabs/proxy-wasm-go-sdk/proxywasm/types
ok      github.com/tetratelabs/proxy-wasm-go-sdk/examples/dispatch_call_on_tick (cached)
ok      github.com/tetratelabs/proxy-wasm-go-sdk/examples/foreign_call_on_tick  (cached)
ok      github.com/tetratelabs/proxy-wasm-go-sdk/examples/helloworld    (cached)
ok      github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_auth_random      (cached)
ok      github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_body     (cached)
--- FAIL: TestHttpHeaders_OnHttpRequestHeaders (0.00s)
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 6 [running]:
testing.tRunner.func1.2({0x5b7d60, 0xc000018360})
        /usr/local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1212 +0x218
panic({0x5b7d60, 0xc000018360})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/proxytest.(*httpHostEmulator).httpHostEmulatorProxyGetHeaderMapValue(0x5c8a19, 0x0, 0x40e987, 0x9, 0xc000010070, 0xc000016ad0)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/proxytest/http.go:141 +0x1a6
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/proxytest.(*hostEmulator).ProxyGetHeaderMapValue(0x4, 0xe22c0, 0x4, 0xc000095cb8, 0x4ffa48, 0xc0000a22a0)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/proxytest/proxytest.go:145 +0x45
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/internal.ProxyGetHeaderMapValue(...)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/internal/abi_hostcalls_mock.go:194
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm.getMapValue(0xed0e0, {0x5c9af1, 0x203000})
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/hostcall.go:745 +0xa4
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm.GetHttpRequestHeader(...)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/hostcall.go:240
github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_headers.(*httpHeaders).OnHttpRequestHeaders(0x70, 0x7f5913777108, 0x2)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/examples/http_headers/main.go:63 +0x6e
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/internal.proxyOnRequestHeaders(0x2, 0x411f05, 0x2)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/internal/abi_callback_l7.go:29 +0x6e
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/internal.ProxyOnRequestHeaders(...)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/internal/abi_callback_test_export.go:53
github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/proxytest.(*httpHostEmulator).CallOnRequestHeaders(0x2, 0x86630, {0xc0000ae540, 0x3, 0x3}, 0x1)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/proxywasm/proxytest/http.go:349 +0xd6
github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_headers.TestHttpHeaders_OnHttpRequestHeaders(0x0)
        /home/mssn/workspace/work/tetrate/oss/proxy-wasm-go-sdk/examples/http_headers/main_test.go:30 +0x165
testing.tRunner(0xc0000d7040, 0x5dbee8)
        /usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1306 +0x35a
FAIL    github.com/tetratelabs/proxy-wasm-go-sdk/examples/http_headers  0.005s
...
```

## Envoy version

```
$ envoy --version
envoy  version: d362e791eb9e4efa8d87f6d878740e72dc8330ac/1.18.2/clean-getenvoy-76c310e-envoy/RELEASE/BoringSSL
```

## What is the SDK version?

upstream HEAD (https://github.com/tetratelabs/proxy-wasm-go-sdk/commit/2041438866a21d52ce7d1c14da7a36ebb48827bc)

## What is your TinyGo version?

```
$ tinygo version
tinygo version 0.20.0 linux/amd64 (using go version go1.17.2 and LLVM version 11.0.0)
```

## Additional context (Optional)

--- a/proxywasm/proxytest/http.go
+++ b/proxywasm/proxytest/http.go
@@ -16,6 +16,7 @@ package proxytest
 
 import (
 	"log"
+	"strings"
 
 	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/internal"
 	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
@@ -137,7 +138,14 @@ func (h *httpHostEmulator) httpHostEmulatorProxyGetHeaderMapValue(mapType intern
 
 	for _, h := range headers {
 		if h[0] == key {
-			value := []byte(h[1])
+			// Leading LWS doesn't affect header values,
+			// and often ignored in HTTP parsers.
+			value := []byte(strings.TrimSpace(h[1]))
+			// If the value is empty,
+			// Envoy ignores scuh headers and return NotFound.
+			if len(value) == 0 {
+				return internal.StatusNotFound
+			}
 			*returnValueData = &value[0]
 			*returnValueSize = len(value)
 			return internal.StatusOK

--- a/proxywasm/proxytest/http.go
+++ b/proxywasm/proxytest/http.go
@@ -142,7 +142,7 @@ func (h *httpHostEmulator) httpHostEmulatorProxyGetHeaderMapValue(mapType intern
 			// and often ignored in HTTP parsers.
 			value := []byte(strings.TrimSpace(h[1]))
 			// If the value is empty,
-			// Envoy ignores scuh headers and return NotFound.
+			// Envoy ignores such headers and return NotFound.
 			if len(value) == 0 {
 				return internal.StatusNotFound
 			}

